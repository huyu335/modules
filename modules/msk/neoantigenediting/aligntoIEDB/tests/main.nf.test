nextflow_process {

    name "Test Process NEOANTIGENEDITING_ALIGNTOIEDB"
    script "../main.nf"
    process "NEOANTIGENEDITING_ALIGNTOIEDB"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "neoantigenediting"
    tag "neoantigenediting/aligntoIEDB"
    tag "ignore_conda"

    test("neoantigenediting_aligntoIEDB - json") {

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['patient_data'], checkIfExists: true)                   
                    ]
                input[1] = [
                    [ id:'fasta', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta'], checkIfExists: true),
                        [
                            file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta_phr'], checkIfExists: true),
                            file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta_pin'], checkIfExists: true),
                            file(params.test_data_mskcc['neoantigen']['iedb']['iedb_fasta_psq'], checkIfExists: true)
                        ]                   
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.iedb_alignment[0][1]).name).match() }

            )
        }

    }

    test("neoantigenediting_aligntoIEDB - json - stub") {

        options "-stub"

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('patient_data')                   
                    ]
                input[1] = [
                    [ id:'fasta', single_end:false ], // meta map,
                    file('iedb_fasta'),
                        [
                            file('iedb_fasta_phr'),
                            file('iedb_fasta_pin'),
                            file('iedb_fasta_psq')
                        ]                   
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }

            )
        }

    }


}
