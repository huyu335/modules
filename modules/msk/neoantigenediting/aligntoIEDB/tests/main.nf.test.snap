{
    "neoantigenediting_aligntoIEDB - json - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "iedb_alignments_example.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c025a6f13374138023afb0416efea864"
                ],
                "iedb_alignment": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "iedb_alignments_example.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c025a6f13374138023afb0416efea864"
                ]
            }
        ],
        "timestamp": "2024-03-19T14:48:40.233961"
    },
    "neoantigenediting_aligntoIEDB - json": {
        "content": [
            "iedb_alignments_3-OLTS.txt"
        ],
        "timestamp": "2024-03-19T14:48:34.589746"
    }
}